src/frequenz/sdk/data_ingestion/resampling/timeseries_resampling_actor.py

```diff
@@ -0,0 +1,159 @@
+                point in time from which to subscribe  for data or `None` to subscribe
+                for data from the live component data stream
+
+        Returns:
+            receiver for receiving samples for a specific time series id
+        """
+        if subscription.time_series_id not in self._output_channels:
+            response: SubscriptionResponse = self._data_provider.subscribe(
+                time_series_id=subscription.time_series_id,
+                # Ignore parquet for now [later => start_time=subscription.start_time]
+                start_time=None,
+            )
+            self._input_receivers[subscription.time_series_id] = response.receiver
+            self._output_channels[subscription.time_series_id] = Broadcast[
+                Optional[Sample]
+            ](str(subscription.time_series_id))
+
+        return self._output_channels[subscription.time_series_id].get_receiver()
+
+    async def run(self) -> None:
+        """Run the actor."""
+        while True:
+            component_data_receivers = {
+                str(time_series_id): receiver
+                for time_series_id, receiver in self._input_receivers.items()
+            }
+
+            select = Select(
+                resampling_timer=self._resampling_timer,
+                subscription_receiver=self._subscription_receiver,
+                component_data_receiver=MergeNamed(**component_data_receivers),
+            )
+            while await select.ready():
+                if msg := select.subscription_receiver:
+                    subscription_request: SubscriptionRequest = msg.inner
+                    sample_receiver: Receiver[Optional[Sample]] = self.subscribe(
+                        subscription=subscription_request.subscription
+                    )
+                    await subscription_request.sender.send(
+                        SubscriptionResponse(
+                            receiver=sample_receiver,
+                            subscription=subscription_request.subscription,
+                        )
+                    )
+                    # Breaking out from the loop is required to regenerate
+                    # component_data_receivers to be able to fulfil this
+                    # subscription (later can be optimized by checking if
+                    # an output channel already existed in the `subscribe()` method)
+                    break
+                if _ := select.resampling_timer:
+                    samples: Dict[
+                        TimeSeriesId, Optional[Sample]
+                    ] = self._resampler.resample()
+                    for time_series_id, sample in samples.items():
+                        await self._output_channels[time_series_id].get_sender().send(
+                            sample
+                        )
+                elif msg := select.component_data_receiver:
+                    name, sample = msg
+                    if sample is not None:
```

Comment:
Oh, there is one edge case we forgot to specify and I think it should be handled here.

AFAIK the microgrid API can return values that are a `float('nan')` (i.e. a [IEEE_754-1985 NaN](https://en.wikipedia.org/wiki/IEEE_754-1985#NaN)).

@tiyash-basu-frequenz please confirm, maybe @sahas-subramanian-frequenz knows too?

We could just use the `NaN` as is in the pipeline, but this is handled super tricky in all programming languages, the most infamous operation IMHO being:

```py
>>> float('nan')
nan
>>> float('nan') is float('nan')
False
>>> float('nan') == float('nan')
False
>>> float('nan') != float('nan')
True
```

So I think we are better off converting `NaN`s to `None` in general, or at this level just discard `NaN` values as if no data would have been arrived. To know if a `float` is a `NaN` we need to use `math.isnan()`.

Reply:
> AFAIK the microgrid API can return values that are a float('nan') (i.e. a [IEEE_754-1985 NaN](https://en.wikipedia.org/wiki/IEEE_754-1985#NaN)).

Yes, it can, since it falls under the specifications of `float`. IMO, it's better to send `None` here instead of `NaN`.